### PR TITLE
Add basic api

### PR DESCRIPTION
**Pull request by AndrejApa**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+)
+
+const (
+	weatherAPI = "http://api.openweathermap.org/data/2.5/weather"
+	apiToken   = "2ae6e50490e35649304462a5d8f6cb29"
```
Comment (AliakseiM) [inline main.go]: It is important not to push your secrets in Github repo. Store it in .env file for example

File: main.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		return
+	}
+	//fmt.Println(string(jitem))
+
+	http.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		_, _ = fmt.Fprintf(w, "%s\n", jitem)
+
+	})
+	fmt.Println("Server is listening...")
+	_ = http.ListenAndServe("127.0.0.1:8080", nil)
```
Comment (AliakseiM) [inline main.go]: Don't ignore errors returned by functions. There may be some important information

**Review by AliakseiM — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (AndrejApa) [reply main.go]: I will take it into account

File: main.go (hunk 1 above)
Comment (AndrejApa) [reply main.go]: fixed

File: main.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+}
+func getWeatherByCity(city string) (WeatherResponse, error) {
+	req, err := http.NewRequest("GET", weatherAPI, nil)
+	if err != nil {
+		return WeatherResponse{}, err
+	}
+	APIKey, exists := os.LookupEnv("API_KEY")
+	if !exists {
+		fmt.Println("Error,see env file")
```
Comment (AliakseiM) [inline main.go]: Does it make sense to continue function's work in case, if "API_KEY" doesn't exist?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		err = json.NewEncoder(w).Encode(data)
+		if err != nil {
+			return
+		}
+
+	})
+	err := http.ListenAndServe("127.0.0.1:8080", nil)
+	if err != nil {
+		return
```
Comment (AliakseiM) [inline main.go:36]: How do you know if there was an error, and what kind?

File: main.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	req.URL.RawQuery = values.Encode()
+
+	cli := http.Client{}
+
+	resp, err := cli.Do(req)
+	if err != nil {
+		return WeatherResponse{}, err
+	}
+
+	var Weather WeatherResponse
```
Comment (AliakseiM) [inline main.go]: I think, that it is not necessary to keep this variable as exported

File: main.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+type WeatherResponse struct {
+	Name string `json:"name"`
+	Main struct {
+		Temp float64 `json:"temp"`
+	} `json:"main"`
+}
+
+func main() {
+	http.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		city := r.URL.Query().Get("q")
```
Comment (AliakseiM) [inline main.go]: I suggest using a clearer name for query parameter

**Review by AliakseiM — DISMISSED** [review]

**Comment by AliakseiM** [comment]
Also, please use more informative commit messages

File: main.go (hunk 3 above)
Comment (AndrejApa) [reply main.go:60]: Change

Comment (AndrejApa) [reply main.go]: fixed

File: main.go (hunk 6 above)
Comment (AndrejApa) [reply main.go:29]: fixed

File: main.go (hunk 5 above)
Comment (AndrejApa) [reply main.go]: fixed

File: main.go (hunk 4 above)
Comment (AndrejApa) [reply main.go:36]: fixed

File: main.go (hunk 3 above)
Comment (AliakseiM) [reply main.go]: I think, it is better to use os.LookupEnv here and check, if there environment variable is set. If not, return error (such packages as "fmt" or "errors" may help you to create a new error)

File: main.go (hunk 4 above)
Comment (AliakseiM) [reply main.go:36]: Nice.
But there is such thing as log.Fatal() - it prints given error, and exits the program

File: main.go (hunk 6 above)
Comment (AndrejApa) [reply main.go]: fixed

File: main.go [hunk 7]
```diff
@@ -24,28 +24,28 @@ type WeatherResponse struct {
...
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-		err = json.NewEncoder(w).Encode(data)
-		if err != nil {
-			return
-		}
-
-	})
+func weather(w http.ResponseWriter, r *http.Request) {
```
Comment (AliakseiM) [inline main.go:27]: Nice!

**Review by AliakseiM — APPROVED** [review]

**Pull request merged**